### PR TITLE
Refactor prep_data/tabular_data into a single notebook

### PR DESCRIPTION
**Pull request by atqy**
*Issue #, if available:*
Skips daily CI

*Description of changes:*
Consolidated a sequential notebook series into one notebook as the content was short enough. Note that I used git mv to rename the notebooks. `git status` detected it but Github didn't pick that up.

*Testing done:*
Ran notebook end-to-end and it works. Also, took it off my personal CI skip list and got a passing CI test. Also updated the links in RST files, seems to work according to this link: https://amazon-sagemaker-examples.readthedocs.io/en/atqy-refactor-tabular-data-actual/prep_data/index.html

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [x] I have tested my notebook(s) and ensured it runs end-to-end
- [x] I have linted my notebook(s) and code using `tox -e black-format,black-nb-format`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prep_data/tabular_data/train_featurize_train_tabular_data.ipynb [hunk 1]
```diff
@@ -0,0 +1,673 @@
+{
```
Comment (jkroll-aws) [inline prep_data/tabular_data/train_featurize_train_tabular_data.ipynb:1]: <p>Add a conclusion section suggesting what to do next now that you've processed the model and trained a model? Such as resources for how to deploy/predict?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3377/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='52'/>

Comment (atqy) [reply prep_data/tabular_data/train_featurize_train_tabular_data.ipynb:1]: As this is a data preparation notebook, the main focus on the data preparation. The model training is really just a verification that our data preparation worked.

Comment (atqy) [reply prep_data/tabular_data/train_featurize_train_tabular_data.ipynb:1]: IMO, prep_data is a weird folder for this reason

**Review by jkroll-aws — APPROVED** [review]

File: prep_data/tabular_data/train_featurize_train_tabular_data.ipynb (hunk 1 above)
Comment (atqy) [reply prep_data/tabular_data/train_featurize_train_tabular_data.ipynb:1]: Yeah, exactly! Like is it our job to showcase SageMaker capabilities or teach actual ML? Definitely worth a wider discussion for sure

**Pull request merged**